File: src/main/java/yuma140902/yumalib/api/McConst.java
package yuma140902.yumalib.api;

/**
 * バニラMinecraftに関する定数
 */
public final class McConst {

	/**
	 * バニラMinecraftのメタデータに関する定数
	 */
	public static final class Meta {
		public static final int
						PLANK_OAK     = 0,
						PLANK_ACACIA  = 4,
						PLANK_BIRCH   = 2,
						PLANK_DARKOAK = 5,
						PLANK_JUNGLE  = 3,
						PLANK_SPRUCE  = 1;
		
		public static final int
						LOG_OAK       = 0,
						LOG_SPRUCE    = 1,
						LOG_BIRCH     = 2,
						LOG_JUNGLE    = 3,
						LOG2_ACACIA   = 0,
						LOG2_DARK_OAK = 1;
		
		public static final int
						STAIRS_EAST  = 0,
						STAIRS_WEST  = 1,
						STAIRS_SOUTH = 2,
						STAIRS_NORTH = 3;
		
		public static final int
						STAIRS_LOWER = 0b0000,
						STAIRS_UPPER = 0b0100;
		
		public static final int
						PILLAR_UP_DOWN     = 0b0000,
						PILLAR_WEST_EAST   = 0b0100,
						PILLAR_NORTH_SOUTH = 0b1000,
						PILLAR_AXIS_NONE   = 0b1100;
		
		public static final int
						PUMPKIN_NORTH = 0,
						PUMPKIN_EAST  = 1,
						PUMPKIN_SOUTH = 2,
						PUMPKIN_WEST  = 3;
		
		public static final int
						LADDER_SOUTH = 2,
						LADDER_NORTH = 3,
						LADDER_EAST  = 4,
						LADDER_WEST  = 5;
		
		public static final int
						CHEST_SOUTH = 2,
						CHEST_NORTH = 3,
						CHEST_EAST  = 4,
						CHEST_WEST  = 5;
		
		public static final int
						SLAB_STONE        = 0,
						SLAB_SANDSTONE    = 1,
						SLAB_ROCKPLANK    = 2,
						SLAB_COBBLESTONE  = 3,
						SLAB_BRICKS       = 4,
						SLAB_STONEBRICKS  = 5,
						SLAB_NETHERBRICKS = 6,
						SLAB_QUARTZ       = 7;
		
		public static final int
						IS_SLAB_UPPER = 0b1000;
		
		public static final int
						IS_STAIRS_UPSIDE_DOWN = 0b0100;
		
		
		/** ポピー */
		public static final int REDFLOWER_POPPY = 0;
		
		/** ヒスイラン */
		public static final int REDFLOWER_BLUE_ORCHID = 1;
		
		/** レンゲソウ */
		public static final int REDFLOWER_ALLIUM = 2;
		
		/** ヒナソウ */
		public static final int REDFLOWER_HOUSTONIA = 3;
		
		public static final int REDFLOWER_RED_TULIP    = 4;
		public static final int REDFLOWER_ORANGE_TULIP = 5;
		public static final int REDFLOWER_WHITE_TULIP  = 6;
		public static final int REDFLOWER_PINK_TULIP   = 7;
		
		/** フランスギク */
		public static final int REDFLOWER_OXEYE_DAISY = 8;
		
		public static final int
						SANDSTONE_NORMAL   = 0,
						SANDSTONE_CHISELED = 1,
						SANDSTONE_CUT      = 2;
		
		public static final int
						STONEBRICK_NORMAL   = 0,
						STONEBRICK_MOSSY    = 1,
						STONEBRICK_CRACKED  = 2,
						STONEBRICK_CHISELED = 3;
	}


	/**
	 * バニラMinecraftのパーティクルに関する定数
	 */
	public static final class Particle {
		public static final String
						HUGE_EXPLOSION    = "hugeexplosion",
						LARGE_EXPLODE     = "largeexplode",
						FIREWORK_SPARK    = "fireworksSpark",
						BUBBLE            = "bubble",
						SUSPENDED         = "suspended",
						DEPTH_SUSPEND     = "depthsuspend",
						TOWN_AURA         = "townaura",
						CRITICAL          = "crit",
						MAGIC_CRITICAL    = "magicCrit",
						SMOKE             = "smoke",
						MOB_SPELL         = "mobSpell",
						MOB_SPELL_AMBIENT = "mobSpellAmbient",
						SPELL             = "spell",
						INSTANT_SPELL     = "instantSpell",
						WITCH_MAGIC       = "witchMagic",
						NOTE              = "note",
						PORTAL            = "portal",
						ENCHANTMENT_TABLE = "enchantmenttable",
						EXPLODE           = "explode",
						FLAME             = "flame",
						LAVA              = "lava",
						FOOTSTEP          = "footstep",
						SPLASH            = "splash",
						WAKE              = "wake",
						LARGE_SMOKE       = "largesmoke",
						CLOUD             = "cloud",
						REDDUST           = "reddust",
						SNOWBALLPOOF      = "snowballpoof",
						DRIP_WATER        = "dripWater",
						DRIP_LAVA         = "dripLava",
						SNOW_SHOVEL       = "snowshovel",
						SLIME             = "slime",
						HEART             = "heart",
						ANGRY_VILLAGER    = "angryVillager",
						HAPPY_VILLAGER    = "happyVillager";
	}
	
	
	public static final int
					SIDE_BOTTOM = 0,
					SIDE_TOP    = 1,
					SIDE_NORTH  = 2,
					SIDE_SOUTH  = 3,
					SIDE_WEST   = 4,
					SIDE_EAST   = 5;
	
	public static final float
					EXP_STONE  = 0.1F,
					EXP_COAL   = 0.15F,
					EXP_CACTUS = 0.2F,
					EXP_BRICK  = 0.3F,
					EXP_MEAT   = 0.35F,
					EXP_IRON   = 0.7F,
					EXP_GEM    = 1.0F;
	
}

File: src/main/scala/yuma140902/yumalib/api/McConst.scala
package yuma140902.yumalib.api

object McConst {

  /**
   * バニラMinecraftのメタデータに関する定数
   */
  object Meta {
    val PLANK_OAK = 0
    val PLANK_ACACIA = 4
    val PLANK_BIRCH = 2
    val PLANK_DARKOAK = 5
    val PLANK_JUNGLE = 3
    val PLANK_SPRUCE = 1

    val LOG_OAK = 0
    val LOG_SPRUCE = 1
    val LOG_BIRCH = 2
    val LOG_JUNGLE = 3
    val LOG2_ACACIA = 0
    val LOG2_DARK_OAK = 1

    val STAIRS_EAST = 0
    val STAIRS_WEST = 1
    val STAIRS_SOUTH = 2
    val STAIRS_NORTH = 3

    val STAIRS_LOWER: Int = 0x0 //0b0000
    val STAIRS_UPPER: Int = 0x4 //0b0100

    val PILLAR_UP_DOWN: Int = 0x0 //0b0000
    val PILLAR_WEST_EAST: Int = 0x4 //0b0100
    val PILLAR_NORTH_SOUTH: Int = 0x8 //0b1000
    val PILLAR_AXIS_NONE: Int = 0xc //0b1100

    val PUMPKIN_NORTH = 0
    val PUMPKIN_EAST = 1
    val PUMPKIN_SOUTH = 2
    val PUMPKIN_WEST = 3

    val LADDER_SOUTH = 2
    val LADDER_NORTH = 3
    val LADDER_EAST = 4
    val LADDER_WEST = 5

    val CHEST_SOUTH = 2
    val CHEST_NORTH = 3
    val CHEST_EAST = 4
    val CHEST_WEST = 5

    val SLAB_STONE = 0
    val SLAB_SANDSTONE = 1
    val SLAB_ROCKPLANK = 2
    val SLAB_COBBLESTONE = 3
    val SLAB_BRICKS = 4
    val SLAB_STONEBRICKS = 5
    val SLAB_NETHERBRICKS = 6
    val SLAB_QUARTZ = 7

    val IS_SLAB_UPPER: Int = 0x8 //0b1000
    val IS_STAIRS_UPSIDE_DOWN: Int = 0x4 //0b0100


    /** ポピー */
    val REDFLOWER_POPPY = 0
    /** ヒスイラン */
    val REDFLOWER_BLUE_ORCHID = 1
    /** レンゲソウ */
    val REDFLOWER_ALLIUM = 2
    /** ヒナソウ */
    val REDFLOWER_HOUSTONIA = 3
    val REDFLOWER_RED_TULIP = 4
    val REDFLOWER_ORANGE_TULIP = 5
    val REDFLOWER_WHITE_TULIP = 6
    val REDFLOWER_PINK_TULIP = 7
    /** フランスギク */
    val REDFLOWER_OXEYE_DAISY = 8

    val SANDSTONE_NORMAL = 0
    val SANDSTONE_CHISELED = 1
    val SANDSTONE_CUT = 2

    val STONEBRICK_NORMAL = 0
    val STONEBRICK_MOSSY = 1
    val STONEBRICK_CRACKED = 2
    val STONEBRICK_CHISELED = 3
  }

  object Particle {
    val HUGE_EXPLOSION = "hugeexplosion"
    val LARGE_EXPLODE = "largeexplode"
    val FIREWORK_SPARK = "fireworksSpark"
    val BUBBLE = "bubble"
    val SUSPENDED = "suspended"
    val DEPTH_SUSPEND = "depthsuspend"
    val TOWN_AURA = "townaura"
    val CRITICAL = "crit"
    val MAGIC_CRITICAL = "magicCrit"
    val SMOKE = "smoke"
    val MOB_SPELL = "mobSpell"
    val MOB_SPELL_AMBIENT = "mobSpellAmbient"
    val SPELL = "spell"
    val INSTANT_SPELL = "instantSpell"
    val WITCH_MAGIC = "witchMagic"
    val NOTE = "note"
    val PORTAL = "portal"
    val ENCHANTMENT_TABLE = "enchantmenttable"
    val EXPLODE = "explode"
    val FLAME = "flame"
    val LAVA = "lava"
    val FOOTSTEP = "footstep"
    val SPLASH = "splash"
    val WAKE = "wake"
    val LARGE_SMOKE = "largesmoke"
    val CLOUD = "cloud"
    val REDDUST = "reddust"
    val SNOWBALLPOOF = "snowballpoof"
    val DRIP_WATER = "dripWater"
    val DRIP_LAVA = "dripLava"
    val SNOW_SHOVEL = "snowshovel"
    val SLIME = "slime"
    val HEART = "heart"
    val ANGRY_VILLAGER = "angryVillager"
    val HAPPY_VILLAGER = "happyVillager";
  }

  val SIDE_BOTTOM = 0
  val SIDE_TOP = 1
  val SIDE_NORTH = 2
  val SIDE_SOUTH = 3
  val SIDE_WEST = 4
  val SIDE_EAST = 5;

  val EXP_STONE = 0.1F
  val EXP_COAL = 0.15F
  val EXP_CACTUS = 0.2F
  val EXP_BRICK = 0.3F
  val EXP_MEAT = 0.35F
  val EXP_IRON = 0.7F
  val EXP_GEM = 1.0F;
}
